staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -782,6 +782,10 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 			options = append(options, clientv3.WithRev(withRev))
 		}
 	}
+	if v.IsNil() {
+		// Ensure that we never return a nil Items pointer in the result for consistency.
```

Comment:
FWIW - I found that when playing with unifying tests - I found that very confusing that when I know that no items are returned, I don't know if the returned result is nil or empty slice - as it depends on what parameters are set (label/field selectors), whether we're using watchcache or etcd3 store and what items are returned underneath from etcd3.

Reply:
This `nil` vs `[]` distinction wasn't surfaced to API clients, right? if I remember correctly, for zero-length list results, protobuf clients always get `nil` (because protobuf serialization does not preserve that distinction), and json clients always get `[]`, right?

Is this solely for the benefit of internal test consistency, or are there non-test code paths internal to the API server that rely on a non-nil value for list results to be correct?

Reply:
> This nil vs [] distinction wasn't surfaced to API clients, right?

Correct. This problem was not exposed to Kubernetes API clients.
This inconsistency was correctly handled at the level of generic registry so the responses send to clients were unified.

This was only potentially visible for users of `storage.Interface`. Fortunately, apart from generic registry, we only have very few internal ones related to internal reconcilers:
- lease reconciler in kube-apiserver
- IP and port allocator for services
And obviously a bunch of test code.

The three mentioned above weren't affected. So as of now, this is pretty much "internal test consistency" change.

Reply:
ok, thanks